source/common/upstream/maglev_lb.cc

```diff
@@ -26,7 +26,7 @@ MaglevTable::MaglevTable(const NormalizedHostWeightVector& normalized_host_weigh
   table_build_entries.reserve(normalized_host_weights.size());
   for (const auto& host_weight : normalized_host_weights) {
     const auto& host = host_weight.first;
-    const std::string& address = host->address()->asString();
+    const std::string& address = use_hostname ? host->hostname() : host->address()->asString();
```

Comment:
After this line can you ASSERT that address is not empty? Same in the ring hash case.